src/OrbitGl/App.cpp

```diff
@@ -2301,6 +2303,7 @@ void OrbitApp::AddFrameTrack(const FunctionInfo& function) {
 
 void OrbitApp::AddFrameTrack(uint64_t instrumented_function_id) {
   CHECK(instrumented_function_id != 0);
+  CHECK(std::this_thread::get_id() == main_thread_id_);
   if (!HasCaptureData()) {
```

Comment:
Maybe put a comment here. // This is only the case when an old capture file was loaded